forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/services/FormSubmissionService.java

```diff
@@ -161,14 +163,24 @@ private String getSubmissionUrl(String formUrl){
     }
 
     public Map<String,Object> retrieveFormValues(String formUrl) throws IOException {
+        return this.retrieveFormValues(formUrl, true);
+    }
+
+    public Map<String,Object> retrieveFormValues(String formUrl, boolean withFileInfo) throws IOException {
         Map<String,Object> fieldValues = new HashMap();
         String submission = readSubmission(formUrl);
         if(StringUtils.isNotEmpty(submission)) {
             JsonNode dataNode = getObjectMapper().readTree(submission);
             Iterator<Map.Entry<String, JsonNode>> dataElements = dataNode.findPath("data").fields();
             while (dataElements.hasNext()) {
                 Map.Entry<String, JsonNode> entry = dataElements.next();
-                if(StringUtils.endsWithIgnoreCase(entry.getKey(),"_file")) {
+
+                if(!withFileInfo && entry.getValue() != null && entry.getValue().isArray() && entry.getValue().get(0).has("originalName")) {
+                    ArrayNode vals = (ArrayNode) entry.getValue();
+                    String fileNames = StreamSupport.stream(vals.spliterator(), false).map(n -> n.get("originalName").asText())
```

Comment:
These are the things I was struggling with on my task.  Not having knowledge of some of the classes/utilities available to deal with different types.  Very cool!

Reply:
Yup, Java streams are indeed pretty powerful, but takes some getting used to! https://stackify.com/streams-guide-java-8/ is a good read if interested